From c1d744fe54aa2c83e0037a1d2a72fdc57163225f Mon Sep 17 00:00:00 2001
From: Fedora GDB patches <invalid@email.com>
Date: Fri, 27 Oct 2017 21:07:50 +0200
Subject: Work around PR libc/13097 "linux-vdso.so.1" #2

FileName: gdb-glibc-vdso-workaround.patch

;; Work around PR libc/13097 "linux-vdso.so.1" warning message.
;;=push+jan

http://sourceware.org/ml/gdb-patches/2011-08/msg00331.html

Hi,

missed the x86_64-m32 case:

gdb/
2011-08-16  Jan Kratochvil  <jan.kratochvil@redhat.com>

	Work around PR libc/13097.
	* solib.c (update_solib_list): Ignore "linux-vdso.so.1".
---
 gdb/solib.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/gdb/solib.c b/gdb/solib.c
index cb7a5c0b51..511671ea9d 100644
--- a/gdb/solib.c
+++ b/gdb/solib.c
@@ -884,8 +884,11 @@ update_solib_list (int from_tty)
 
 	  TRY
 	    {
-	      /* Fill in the rest of the `struct so_list' node.  */
-	      if (!solib_map_sections (i))
+	      /* Fill in the rest of the `struct so_list' node.
+		 Work around PR libc/13097.  */
+	      if (!solib_map_sections (i)
+		  && strcmp (i->so_original_name, "linux-vdso.so.1") != 0
+		  && strcmp (i->so_original_name, "linux-gate.so.1") != 0)
 		{
 		  not_found++;
 		  if (not_found_filename == NULL)
-- 
2.14.3

